"""
This part of the workflow collects the phylogenetic tree and annotations to
export a reference tree and create the Nextclade dataset.

REQUIRED INPUTS:

    TODO: Confirm inputs for Nextclade v3
    reference           = ../shared/reference.fasta
    pathogen            = config/pathogen.json
    genome_annotation   = config/genome_annotation.gff3
    readme              = config/README.md
    changelog           = config/CHANGELOG.md
    metadata            = data/metadata.tsv
    tree                = results/tree.nwk
    branch_lengths      = results/branch_lengths.json
    nt_muts             = results/nt_muts.json
    aa_muts             = results/aa_muts.json
    clades              = results/clades.json


OUTPUTS:

    nextclade_dataset = datasets/${build_name}/*

    See Nextclade docs on expected naming conventions of dataset files
    https://docs.nextstrain.org/projects/nextclade/page/user/datasets.html

This part of the workflow usually includes the following steps:

    - augur export v2
    - cp Nextclade datasets files to new datasets directory

See Augur's usage docs for these commands for more details.
"""
